library/src/main/java/org/xmtp/android/library/libxmtp/GroupEmitter.kt

```diff
@@ -0,0 +1,17 @@
+package uniffi.xmtpv3.org.xmtp.android.library.libxmtp
+
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.asSharedFlow
+import uniffi.xmtpv3.FfiConversationCallback
+import uniffi.xmtpv3.FfiGroup
+
+class GroupEmitter {
+    private val _groups = MutableSharedFlow<FfiGroup>()
+    val groups = _groups.asSharedFlow()
+
+    val callback: FfiConversationCallback = object : FfiConversationCallback {
+        override fun onConversation(conversation: FfiGroup) {
+            _groups.tryEmit(conversation)
```

Comment:
If you're having trouble with streaming, might be worth logging something here so we can isolate if the problem is in Rust or Kotlin. Totally possible something is locking up in Rust